vmaas_sync/vmaas_sync.py

```diff
@@ -240,12 +240,70 @@ def _websocket_connect_status(self, future):
 
         self.vmaas_websocket = result
 
-    async def re_evaluate_systems(self):
+    @staticmethod
+    def select_repo_based_inventory_ids(cur, repos: list):
+        """Select inventory-ids connected with inserted repos, don't fetch it."""
+        if repos:
+            cur.execute("""select distinct inventory_id from system_repo where repo_id in
+                           (select id from repo where name in %s)""", (tuple(repos),))
+        else:
+            cur.execute("""select * from system_repo where (1=0)""")  # ensure empty result
+
+    @staticmethod
+    def select_all_inventory_ids(cur):
+        """Select all inventory-ids, don't fetch it."""
+        cur.execute("select inventory_id from system_platform")
+
+    @staticmethod
+    def get_last_repobased_eval_tms(cur):
+        """Select last repo-based evaluation timestamp."""
+        cur.execute("select value from timestamp_kv where name = %s", (constants.TIMESTAMP_LAST_REPO_BASED_EVAL,))
+        ret = cur.fetchone()
+        if ret:
+            return ret[0]
+        return None
+
+    @staticmethod
+    def set_last_repobased_eval_tms(cur, timestamp: dt.datetime):
+        """Update last repo-based evaluation timestamp."""
+        cur.execute("""insert into timestamp_kv (name, value) values (%s, %s)
+                    on conflict (name) do update set value = %s returning value, (xmax = 0) as inserted""",
+                    (constants.TIMESTAMP_LAST_REPO_BASED_EVAL, timestamp, timestamp))
+        ret = cur.fetchone()
+        return ret
+
+    @staticmethod
+    def _vmaas_repos_modified_since(modified_since: str) -> list:
+        repos_json = {'repository_list': ['.*'], 'modified_since': modified_since}
+        r_json = utils.vmaas_post_request(VMAAS_REPOS_ENDPOINT, repos_json, requests.Session())
+        repos = [repo_name_key for repo_name_key in r_json['repository_list']]
+        LOGGER.info("%d repos found updated since %s", len(repos), modified_since)
+        return repos
+
+    def _get_updated_repos(self, conn) -> list:
+        """Get repos updated since last repo-based evaluation"""
+        with conn.cursor() as cur:
+            modified_since_dt = self.get_last_repobased_eval_tms(cur)
+            if modified_since_dt is None:
+                modified_since_dt = dt.datetime.utcfromtimestamp(0).replace(tzinfo=dt.timezone.utc)
+            repos = self._vmaas_repos_modified_since(modified_since_dt.isoformat())
+            self.set_last_repobased_eval_tms(cur, dt.datetime.now())
+            return repos
+
+    async def re_evaluate_systems(self, repo_based: bool):
         """Schedule re-evaluation for all systems in DB."""
-        LOGGER.info("Re-evaluating all systems")
+        LOGGER.info("Re-evaluating")
```

Comment:
maybe delete this message because later it's printed better based on repo_based value